solutions/agents/variables.tf

```diff
@@ -0,0 +1,222 @@
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud api key."
+  sensitive   = true
+}
+
+##############################################################################
+# Cluster variables
+##############################################################################
+
+variable "cluster_id" {
+  type        = string
+  description = "The ID of the cluster you wish to deploy the agents in."
+}
+
+variable "cluster_resource_group_id" {
+  type        = string
+  description = "The resource group ID of the cluster."
+}
+
+variable "cluster_config_endpoint_type" {
+  description = "Specify which type of endpoint to use for for cluster config access: 'default', 'private', 'vpe', 'link'. 'default' value will use the default endpoint of the cluster."
+  type        = string
+  default     = "private"
+  nullable    = false # use default if null is passed in
+  validation {
+    error_message = "Invalid Endpoint Type! Valid values are 'default', 'private', 'vpe', or 'link'"
+    condition     = contains(["default", "private", "vpe", "link"], var.cluster_config_endpoint_type)
+  }
+}
+
+##############################################################################
+# Log Analysis variables
+##############################################################################
+
+variable "log_analysis_enabled" {
+  type        = bool
+  description = "Deploy IBM Cloud Logging agent if set as true."
+  default     = true
+}
+
+
+variable "log_analysis_agent_tags" {
+  type        = list(string)
+  description = "List of tags to associate to all log records that the agent collects so that you can identify the agent's data quicker in the logging UI. NOTE: Use the 'log_analysis_add_cluster_name' variable to add the cluster name as a tag."
+  default     = []
+  nullable    = false
+}
+
+variable "log_analysis_add_cluster_name" {
+  type        = bool
+  description = "If true, configure the log analysis agent to attach a tag containing the cluster name to all log messages."
+  default     = true
+}
+
+variable "log_analysis_ingestion_key" {
+  type        = string
+  description = "Ingestion key for the IBM Cloud Logging agent to communicate with the instance."
+  sensitive   = true
+  default     = null
+}
+
+variable "log_analysis_secret_name" {
+  type        = string
+  description = "The name of the secret which will store the ingestion key."
+  default     = "logdna-agent"
+  nullable    = false
+}
+
+variable "log_analysis_instance_region" {
+  type        = string
+  description = "The IBM Log Analysis instance region. Used to construct the ingestion endpoint."
+  default     = "us-south"
```

Comment:
I would suggest having no default value for this, so its obvious to the consumer they need to add the region that their instance is in

Reply:
Ah but I guess its not required if `log_analysis_enabled` is set to false. Hmm tricky use case to handle. You could default it to `null` here and add variable validation to ensure a value is passed if `log_analysis_enabled` is set to true. My worry is that the user will deploy the agents and the pod wont come up because its hitting the wrong regional endpoint for the ingestion key they provided.

Reply:
variable validation is there in root (observability-agents) module -
`log_analysis_key_validate_condition = var.log_analysis_enabled == true && var.log_analysis_instance_region == null && var.log_analysis_ingestion_key == null`
So validation part is taken care already.

Reply:
Regarding instance_region --> how about fetching it from a variable which instance DA outputs or user can get easily from the existing log analysis....? I am thinking of fetching it from instance_crn

Reply:
@maheshwarishikha Are you talking about updating the DA to take in an instance CRN? This is actually how the module used to work but we had to change it because its possible the instance will not exist in the same account as the cluster where you are deploying the agent, so hence you can't do a data lookup on the instance to get its region.

Reply:
One thing we could do is mark all the region related variables as required in catalog, so they show up in the required variable tab in Projects and its easier for a consumer to see. Its something we do in landing zone right now with the SSH key input. I'll create a follow up issue for that, because there may be a few modules we will want to do it with. Lets just leave us-south as default for now

Reply:
defaulted to `null` now.